astronomy/orbit_analysis_test.cpp

```diff
@@ -0,0 +1,509 @@
+                /*step=*/1 * JulianYear))),
+        earth_(*earth_1957_.rotating_body(*ephemeris_, "Earth")) {}
+
+  // Returns a GCRS trajectory obtained by stitching together the trajectories
+  // of |sp3_orbit.satellites| in |sp3_orbit.files|.
+  not_null<std::unique_ptr<DiscreteTrajectory<GCRS>>> EarthCentredTrajectory(
+      SP3Orbit const& sp3_orbit) {
+    BodyCentredNonRotatingDynamicFrame<ICRS, GCRS> gcrs{ephemeris_.get(),
+                                                        &earth_};
+    BodySurfaceDynamicFrame<ICRS, ITRS> itrs{ephemeris_.get(), &earth_};
+
+    auto result = make_not_null_unique<DiscreteTrajectory<GCRS>>();
+    for (auto const& file : sp3_orbit.files.names) {
+      StandardProduct3 sp3(
+          SOLUTION_DIR / "astronomy" / "standard_product_3" / file,
+          sp3_orbit.files.dialect);
+      auto const& orbit = sp3.orbit(sp3_orbit.satellite);
+      CHECK_EQ(orbit.size(), 1);
+      auto const& arc = *orbit.front();
+      for (auto it = arc.Begin(); it != arc.End(); ++it) {
+        ephemeris_->Prolong(it.time());
+        result->Append(
+            it.time(),
+            gcrs.ToThisFrameAtTime(it.time())(
+                itrs.FromThisFrameAtTime(it.time())(it.degrees_of_freedom())));
+      }
+    }
+    return result;
+  }
+
+  std::pair<OrbitalElements, OrbitRecurrence> ElementsAndRecurrence(
+      SP3Orbit const& orbit) {
+    auto const elements =
+        OrbitalElements::ForTrajectory(*EarthCentredTrajectory(orbit),
+                                       earth_,
+                                       MasslessBody{}).ValueOrDie();
+    auto const recurrence =
+        OrbitRecurrence::ClosestRecurrence(elements.nodal_period(),
+                                           elements.nodal_precession(),
+                                           earth_,
+                                           /*max_abs_Cᴛₒ=*/100);
+    return {elements, recurrence};
+  }
+
+  SolarSystem<ICRS> earth_1957_;
+  not_null<std::unique_ptr<Ephemeris<ICRS>>> ephemeris_;
+  RotatingBody<ICRS> const& earth_;
+
+ private:
+  SolarSystem<ICRS> RemoveAllButEarth(SolarSystem<ICRS> solar_system) {
+    std::vector<std::string> const names = solar_system.names();
+    for (auto const& name : names) {
+      if (name != "Earth") {
+        solar_system.RemoveMassiveBody(name);
+      }
+    }
+    return solar_system;
+  }
+};
+
+// COSPAR ID 2010-001A, SVN C003.
+// 北斗二號 GEO01.
+// PRN C01, GEO, 140.0° E.
+TEST_F(OrbitAnalysisTest, 北斗GEO) {
+  auto [elements, recurrence] = ElementsAndRecurrence(  // NOLINT
+      {{StandardProduct3::SatelliteGroup::北斗, 1}, SP3Files::GNSS()});
+
+  EXPECT_THAT(recurrence,
+              AllOf(Property(&OrbitRecurrence::νₒ, 1),
+                    Property(&OrbitRecurrence::Dᴛₒ, 0),
+                    Property(&OrbitRecurrence::Cᴛₒ, 1)));
+  EXPECT_THAT(elements.mean_semimajor_axis_interval().midpoint(),
+              IsNear(42'166 * Kilo(Metre)));
```

Comment:
Do these numbers come from some reference document?  Can you give references?  Same questions everywhere.

Reply:
I give references where they do come from some reference document; where they do not, they are just the truncation of whatever the test spat out.

This will be made clearer once we have the new matchers: we can use
```
12'345_⑴ * Kilo(Metre)
```
for the change-detector tests, and
```C++
AbsoluteErrorFrom(12'345 * Kilo(Metre), IsNear(3.2_⑴ * Kilo(Metre)))
```
where we have nominal values.